core/src/main/scala/chisel3/internal/Binding.scala

```diff
@@ -9,179 +9,175 @@ import chisel3.properties.Class
 
 import scala.collection.immutable.VectorMap
 
-// Element only direction used for the Binding system only.
-private[chisel3] sealed abstract class BindingDirection
-private[chisel3] object BindingDirection {
-
-  /** Internal type or wire
-    */
-  case object Internal extends BindingDirection
-
-  /** Module port with output direction
-    */
-  case object Output extends BindingDirection
-
-  /** Module port with input direction
-    */
-  case object Input extends BindingDirection
-
-  /** Determine the BindingDirection of an Element given its top binding and resolved direction.
-    */
-  def from(binding: TopBinding, direction: ActualDirection): BindingDirection = {
-    binding match {
-      case _: PortBinding | _: SecretPortBinding =>
-        direction match {
-          case ActualDirection.Output => Output
-          case ActualDirection.Input  => Input
-          case dir                    => throw new RuntimeException(s"Unexpected port element direction '$dir'")
-        }
-      case _ => Internal
+private[chisel3] object binding {
+
+  // Element only direction used for the Binding system only.
+  sealed abstract class BindingDirection
+  object BindingDirection {
+
+    /** Internal type or wire
+      */
+    case object Internal extends BindingDirection
+
+    /** Module port with output direction
+      */
+    case object Output extends BindingDirection
+
+    /** Module port with input direction
+      */
+    case object Input extends BindingDirection
+
```

Comment:
This diff looks crazy, but all it is is adding `private[chisel3] object binding {`, tabbing everything over, and deleting all nested `private[chisel3]` and deprecation messages. It's split into 2 commits that at least makes that more clear in the normal commit history. I'll merge with a merge commit.